### PR TITLE
feat: VolumeSnapshot and VolumeSnapshotClass implement

### PR DESCRIPTION
**Pull request by PaulYuuu**
Update VolumeSnapshotClass class, add DeletionPolicy type, also add support for resource creation.

Refs: https://kubernetes.io/docs/concepts/storage/volume-snapshot-classes

##### Short description:
Add support for VolumeSnapshotClass resource creation

##### More details:
The current code does not support creating VolumeSnapshotClass, only pointing to the api_group.

##### What this PR does / why we need it:
This PR aims to support VolumeSnapshotClass creation. It added the DeletionPolicy type and supports key arguments during class initialization, referring to https://kubernetes.io/docs/concepts/storage/volume-snapshot-classes.

## Summary by CodeRabbit

- **New Features**
	- Enhanced `VolumeSnapshotClass` with additional optional parameters: `deletion_policy`, `driver`, and `parameters`.
	- Introduced `VolumeSnapshot` class with new parameters: `source` and `volume_snapshot_class_name` for improved snapshot management.

- **Bug Fixes**
	- Improved error handling for required parameters in the `to_dict` method, ensuring better user experience.

### CODE REVIEW

**Comment by rh-bot-1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install
 * `/retest all`: Retest all

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Review by dbasunag — COMMENTED** [review]
/lgtm

**Review by jpeimer — APPROVED** [review]

File: ocp_resources/volume_snapshot_class.py [hunk 1]
```diff
@@ -7,3 +7,47 @@ class VolumeSnapshotClass(Resource):
...
+        """
+
+        DELETE = "Delete"
+        RETAIN = "Retain"
+
+    def __init__(
+        self,
+        driver=None,
+        deletion_policy=None,
+        parameters=None,
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/volume_snapshot_class.py]: This is generated code, no need to add validation.

File: ocp_resources/volume_snapshot_class.py [hunk 2]
```diff
@@ -7,3 +7,47 @@ class VolumeSnapshotClass(Resource):
...
+        **kwargs,
+    ):
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/volume_snapshot_class.py]: ditto

**Comment by myakove** [comment]
@coderabbitai pause

**Review by myakove — APPROVED** [review]

**Comment by redhat-qe-bot2** [comment]
No can-be-merged configured for this repository

**Pull request merged**